driver-core/src/test/java/com/datastax/driver/core/WarningsTest.java

```diff
@@ -49,5 +69,80 @@ public void should_expose_warnings_on_execution_info() {
 
     List<String> warnings = rs.getExecutionInfo().getWarnings();
     assertThat(warnings).hasSize(1);
+    // also assert that by default, the warning is logged and truncated to MAX_QUERY_LOG_LENGTH
+    String log = logAppender.getNext();
+    assertThat(log).isNotNull();
+    assertThat(log).isNotEmpty();
+    assertThat(log)
+        .isEqualTo(
+            "Query '"
+                + "BEGIN UNLOGGED BATCH\nINSERT INTO foo (k, v) VALUES"
+                + "' generated server side warning(s): "
+                + "Batch for [ks_1.foo] is of size 5152, exceeding specified threshold of 5120 by 32."
+                + Layout.LINE_SEP);
+  }
+
+  @Test(groups = "short")
+  public void should_execute_query_and_log_server_side_warnings() {
+    // Assert that logging of server-side query warnings is NOT disabled
+    assertThat(Boolean.getBoolean(RequestHandler.DISABLE_QUERY_WARNING_LOGS)).isFalse();
+
+    // Given a query that will produce server side warnings that will be embedded in the
+    // ExecutionInfo
+    final String query = "SELECT count(*) FROM foo";
+    SimpleStatement statement = new SimpleStatement(query);
+    // When the query is executed
+    ResultSet rs = session().execute(statement);
+    // Then the result has 1 Row
+    Row row = rs.one();
+    assertThat(row).isNotNull();
+    // And there is a server side warning captured in the ResultSet's ExecutionInfo
+    ExecutionInfo ei = rs.getExecutionInfo();
+    List<String> warnings = ei.getWarnings();
+    assertThat(warnings).isNotEmpty();
+    assertThat(warnings.size()).isEqualTo(1);
+    assertThat(warnings.get(0)).isEqualTo("Aggregation query used without partition key");
+    // And the driver logged the server side warning
+    String log = logAppender.getNext();
+    assertThat(log).isNotNull();
+    assertThat(log).isNotEmpty();
+    assertThat(log)
+        .isEqualTo(
+            "Query '"
+                + query
+                + "' generated server side warning(s): Aggregation query used without partition key"
+                + Layout.LINE_SEP);
+  }
+
+  @Test(groups = "short")
+  public void should_execute_query_and_not_log_server_side_warnings() {
+    // Get the system property value for disabling logging server side warnings
+    final String disabledLogFlag =
+        System.getProperty(RequestHandler.DISABLE_QUERY_WARNING_LOGS, "false");
+    // assert that logs are NOT disabled
+    assertThat(disabledLogFlag).isEqualTo("false");
+    // Disable the logs
+    System.setProperty(RequestHandler.DISABLE_QUERY_WARNING_LOGS, "true");
```

Comment:
Usually when system properties need to be modified, we use the special test group `isolated`.